Initial support for AWS Batch Job Definitions

jonmer85 opened the pull request:
SUMMARY

This adds automation support for the following AWS Batch Job Definitions
ISSUE TYPE

Feature Pull Request
COMPONENT NAME

amazon module

ANSIBLE VERSION

ansible 2.2.2.0
config file =
configured module search path = Default w/o overrides

jonmeransonos committed:
Initial support for AWS Batch Job Definitions

ryansb commented on lib/ansible/modules/cloud/amazon/batch_job_definition.py:
```diff
@@ -0,0 +1,526 @@
+
+DOCUMENTATION = '''
+---
+module: batch_job_definition
```
Same for this one: since there's a flat namespace, please prefix this as `aws_batch_job_definition`.

supreethSonos replied:
fixed

ryansb commented on lib/ansible/modules/cloud/amazon/batch_job_definition.py:
```diff
@@ -0,0 +1,526 @@
+# logger.setLevel(logging.DEBUG)
+
+
+class AWSConnection:
```
Since this is only ever used once, I think it would be easier to read to just create the different client/resource connections and pass those in.

supreethSonos committed:
Changes include:
	1. Renamed file from batch_job_definition.py to aws_batch_job_definition.py
	2. Fixed ANSIBLE_METADATA version
	3. Updated the RETURN documentation
	4. Used Try Except ImportError block while importing non standard python libraries
	5. Updated the remove_job_definition() method to use deregister_job_definition()
	6. Updated set_api_params() method
	7. Removed the response getting embedded in ansible_facts.
	8. Updated main() to convert the final result to snake_dict using camel_dict_to_snake_dict() to maintain consistency.
	9. Added region and job_definition_arn in argument_spec

supreethSonos committed:
Fixed Botocore error

supreethSonos committed:
Revert "Fixed Botocore error"

This reverts commit 0db6ddef5ba00d4cd063eb3f13243189c0ae9ad2.

s-hertel committed:
Update copyright

Add traceback to exceptions

Remove exception handling around the connection since boto3_conn handles it

Add ec2 documentation fragment

Region is not required because it may be provided as an environment variable

Remove default=None from the arg spec since it is the default and cannot be used with required=true

Add aliases for region

Document job_definition_arn

ryansb commented on lib/ansible/modules/cloud/amazon/aws_batch_job_definition.py:
```diff
@@ -0,0 +1,526 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Jon Meran <jonathan.meran@sonos.com>
```
Should follow copyright guideline http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#copyright and have author info in module author field.

s-hertel committed:
Import shared code

s-hertel committed:
Move set_api_params to module_utils

ryansb commented:
Merged as 05f765bda0
